### PR TITLE
Trying out typescript 2.1 changes

### PR DESCRIPTION
**Pull request by abchatra**
Let's see if this passes build on typescript 1.8

### CODE REVIEW

File: pxtlib/emitter/service.ts [hunk 1]
```diff
@@ -269,7 +269,7 @@ namespace ts.pxtc {
                 pkg,
                 extendsTypes,
                 retType: kind == SymbolKind.Module ? "" : typeOf(decl.type, decl, hasParams),
-                parameters: !hasParams ? null : (decl.parameters || []).map(p => {
+                parameters: (!hasParams || !decl.parameters) ? null : decl.parameters.map((p : any)  => {
```
Comment (riknoll) [inline pxtlib/emitter/service.ts]: Don't need `any` here

File: pxtlib/hf2.ts [hunk 2]
```diff
@@ -101,14 +101,14 @@ namespace pxt.HF2 {
     const HF2_STATUS_EXEC_ERR = 0x02
 
 
-    export function write32(buf: ArrayLike<number>, pos: number, v: number) {
+    export function write32(buf: any, pos: number, v: number) {
```
Comment (riknoll) [inline pxtlib/hf2.ts]: Try `{ [index: number]: number }` instead of `any`

File: pxtlib/hf2.ts [hunk 3]
```diff
@@ -101,14 +101,14 @@ namespace pxt.HF2 {
...
         buf[pos + 0] = (v >> 0) & 0xff;
         buf[pos + 1] = (v >> 8) & 0xff;
         buf[pos + 2] = (v >> 16) & 0xff;
         buf[pos + 3] = (v >> 24) & 0xff;
     }
 
-    export function write16(buf: ArrayLike<number>, pos: number, v: number) {
+    export function write16(buf: any, pos: number, v: number) {
```
Comment (riknoll) [inline pxtlib/hf2.ts]: Try `{ [index: number]: number }` instead of `any`

File: webapp/src/app.tsx [hunk 4]
```diff
@@ -1742,9 +1742,9 @@ function showIcons() {
 function assembleCurrent() {
     compiler.compileAsync({ native: true })
         .then(() => compiler.assembleAsync(getEditor().editorFile.content))
-        .then(v => {
+        .then((v) => {
```
Comment (riknoll) [inline webapp/src/app.tsx]: Nitpick: No paraentheses

File: webapp/src/compiler.ts [hunk 5]
```diff
@@ -27,7 +27,7 @@ function setDiagnostics(diagnostics: pxtc.KsDiagnostic[]) {
     for (let diagnostic of diagnostics) {
         if (diagnostic.fileName) {
             output += `${diagnostic.category == ts.DiagnosticCategory.Error ? lf("error") : diagnostic.category == ts.DiagnosticCategory.Warning ? lf("warning") : lf("message")}: ${diagnostic.fileName}(${diagnostic.line + 1},${diagnostic.column + 1}): `;
-            let f = mainPkg.filterFiles(f => f.getTypeScriptName() == diagnostic.fileName)[0]
+            let f = mainPkg.filterFiles((f: any) => f.getTypeScriptName() == diagnostic.fileName)[0]
```
Comment (riknoll) [inline webapp/src/compiler.ts]: Avoid using `any` in this file.

File: webapp/src/app.tsx [hunk 6]
```diff
@@ -290,7 +290,7 @@ export class ProjectView
         () => {
             let state = this.editor.snapshotState()
             compiler.typecheckAsync()
-                .done(resp => {
+                .done((resp: any) => {
```
Comment (riknoll) [inline webapp/src/app.tsx]: Avoid `any` here

File: webapp/src/app.tsx [hunk 7]
```diff
@@ -566,7 +566,7 @@ export class ProjectView
                 }
 
                 pkg.mainPkg.getCompileOptionsAsync()
-                    .catch(e => {
+                    .catch((e: any) => {
```
Comment (riknoll) [inline webapp/src/app.tsx]: Avoid `any` here

File: pxtlib/emitter/service.ts [hunk 8]
```diff
@@ -695,7 +695,7 @@ namespace ts.pxtc.service {
     let lastBlocksInfo: BlocksInfo;
     let lastFuse: Fuse;
     let builtinItems: SearchInfo[];
-    let tbSubset: Map<boolean>;
+    let tbSubset: pxt.Map<boolean>;
```
Comment (riknoll) [inline pxtlib/emitter/service.ts]: Why not `MapLike`? Also on line 709.

File: cli/cli.ts [hunk 9]
```diff
@@ -3663,7 +3663,7 @@ function extractAsyncInternal(filename: string, out: string, vscode: boolean): P
         .then(dirs => {
             if (dirs && vscode) {
                 pxt.debug('launching code...')
-                dirs.forEach(dir => openVsCode(dir));
+                dirs.forEach((dir: any) => openVsCode(dir));
```
Comment (riknoll) [inline cli/cli.ts]: avoid `any`

File: cli/nodeutil.ts [hunk 10]
```diff
@@ -204,7 +202,7 @@ function nodeHttpRequestAsync(options: Util.HttpRequestOptions): Promise<Util.Ht
         u.headers['content-length'] = buf.length
 
     return new Promise<Util.HttpResponse>((resolve, reject) => {
-        let req = mod.request(u, res => {
+        let callback = (res?: any) => {
```
Comment (riknoll) [inline cli/nodeutil.ts]: avoid `any`

File: cli/server.ts [hunk 11]
```diff
@@ -444,7 +444,7 @@ function initSocketServer(wsPort: number) {
                             case "talk":
                                 return Promise.mapSeries(msg.arg.cmds, (obj: any) =>
                                     hio.talkAsync(obj.cmd, U.fromHex(obj.data))
-                                        .then(res => ({ data: U.toHex(res) })))
+                                        .then((res: any) => ({ data: U.toHex(res) })))
```
Comment (riknoll) [inline cli/server.ts]: avoid `any`

File: webapp/src/monaco.tsx [hunk 12]
```diff
@@ -70,7 +70,7 @@ export class Editor extends srceditor.Editor {
 
         let promise = Promise.resolve().then(() => {
             if (!this.hasBlocks())
-                return
+                return null;
```
Comment (riknoll) [inline webapp/src/monaco.tsx]: Why not `undefined`?

File: webapp/src/monaco.tsx [hunk 13]
```diff
@@ -115,7 +115,7 @@ export class Editor extends srceditor.Editor {
                         }
                     }
                     return compiler.decompileAsync(this.currFile.name, blocksInfo, oldWorkspace, blockFile)
-                        .then(resp => {
+                        .then((resp: any) => {
```
Comment (riknoll) [inline webapp/src/monaco.tsx]: avoid `any`

File: pxtlib/emitter/service.ts (hunk 8 above)
Comment (abchatra) [inline pxtlib/emitter/service.ts]: Shall address differently. Not related to this change. We have MapLike & pxt.Map in our codebase

File: webapp/src/app.tsx (hunk 6 above)
Comment (abchatra) [inline webapp/src/app.tsx]: typecheckAsync returns Promise<any>. I won't fix this now.

File: webapp/src/app.tsx (hunk 7 above)
Comment (abchatra) [inline webapp/src/app.tsx]: Fixed

File: webapp/src/app.tsx (hunk 4 above)
Comment (abchatra) [inline webapp/src/app.tsx]: Done

File: webapp/src/compiler.ts (hunk 5 above)
Comment (abchatra) [inline webapp/src/compiler.ts]: Fixed

File: cli/nodeutil.ts (hunk 10 above)
Comment (abchatra) [inline cli/nodeutil.ts]: Done

File: webapp/src/monaco.tsx (hunk 12 above)
Comment (abchatra) [inline webapp/src/monaco.tsx]: Fixed

File: webapp/src/monaco.tsx (hunk 13 above)
Comment (abchatra) [inline webapp/src/monaco.tsx]: Fixed

File: cli/server.ts (hunk 11 above)
Comment (abchatra) [inline cli/server.ts]: Promise returns any. Won't fix for now

File: cli/cli.ts (hunk 9 above)
Comment (abchatra) [inline cli/cli.ts]: Done

**Comment by pelikhan** [comment]
We can probably close this.

**Pull request closed without merging**